src/ngraph_encapsulate_op.cc

```diff
@@ -291,20 +273,36 @@ class NGraphEncapsulateOp : public OpKernel {
 
     signature_ss << "/";
 
-    std::vector<const Tensor*> static_input_map(ctx->num_inputs());
+    static_input_map.resize(ctx->num_inputs());
     for (int i = 0; i < ctx->num_inputs(); i++) {
       const Tensor& input_tensor = ctx->input(i);
       if (m_input_is_static[i]) {
         static_input_map[i] = &input_tensor;
-        OP_REQUIRES_OK(ctx, TensorToStream(signature_ss, input_tensor));
+        TF_RETURN_IF_ERROR(TensorToStream(signature_ss, input_tensor));
         signature_ss << ";";
       }
     }
+    return Status::OK();
+  }
+
+  Status GetNgExec(OpKernelContext* ctx,
+                   std::shared_ptr<ngraph::runtime::Executable>& ng_exec,
+                   std::vector<TensorShape>& input_shapes,
+                   std::vector<const Tensor*>& static_input_map) {
+    std::stringstream signature_ss;
+    string signature;
 
     std::shared_ptr<ngraph::Function> ng_function;
-    std::shared_ptr<ngraph::runtime::Executable> ng_exec;
     std::shared_ptr<ngraph::runtime::Executable> evicted_ng_exec;
-    std::string signature = signature_ss.str();
+
+    NGRAPH_VLOG(4) << "GetNgExec: Got backend of type: " << m_op_backend_name;
+    ng::runtime::Backend* op_backend =
```

Comment:
Please pass the backend pointer also. We have already retrieved the backend pointer from the backend manager in the compute function. It can be passed to GetNgExec() function.